src/main/java/br/com/alura/cibus/controller/CozinhaController.java

```diff
@@ -0,0 +1,107 @@
+import br.com.alura.cibus.modelo.Cozinha;
+import br.com.alura.cibus.repository.CozinhaRepository;
+
+@Controller
+public class CozinhaController implements ErrorController {
+
+    private CozinhaRepository cozinhaRepository;
+
+    public CozinhaController(CozinhaRepository cozinhaRepository) {
+        this.cozinhaRepository = cozinhaRepository;
+    }
+    
+    @RequestMapping("/error")
+    public String handleError() {
+    	return "/error/not-found";
+    }
+
+    @GetMapping("/cozinhas")
+    public String listar(Model model) {
+        List<Cozinha> listaCozinhas = this.cozinhaRepository.findByOrderByNome();
+        model.addAttribute("listaCozinhas", listaCozinhas);
+
+        return "/cozinhas/listagem";
+    }
+
+    @GetMapping("/admin/cozinhas/novo")
+    public String adicionar() {
+        return "/admin/cozinhas/adicionar";
+    }
+
+    @PostMapping("/admin/cozinhas/salvar")
+    public String salvar(@Valid CozinhaForm form, BindingResult result, Model model) {
+    	if (result.hasFieldErrors("nome")) return "/admin/cozinhas/adicionar";
+        
+        if (cozinhaRepository.existsByNomeIgnoreCase(form.getNome())) {
+        	model.addAttribute("erro", "Já existe uma cozinha com esse nome.");
+        	return "/admin/cozinhas/adicionar";
+        }
+        
+        this.cozinhaRepository.save(form.toCozinha());
+        return "redirect:/cozinhas";
+    }
+    
+    @GetMapping("/admin/cozinhas/editar/{id}")
+    public String editar(@PathVariable String id, Model model) {
+    	Optional<Cozinha> cozinha = this.cozinhaRepository.findById(Long.parseLong(id));
+    	if(cozinha.isEmpty()) {
+    		throw new NotFoundException("Ocorreu um erro. A cozinha não foi encontrada.");
+    	}
+    	
+    	model.addAttribute("cozinha", cozinha.get());
+    	return "/admin/cozinhas/editar";
+    }
+
+    @PostMapping("/admin/cozinhas/salvar/{id}")
+    public String salvar(@PathVariable("id") String id, @Valid CozinhaForm form, BindingResult result, Model model) {
+    	if(result.hasFieldErrors("nome")) return editar(id, model);
+    	
+        Optional<Cozinha> cozinhaPorId = this.cozinhaRepository.findById(Long.parseLong(id));
+        if(cozinhaPorId.isEmpty()) throw new NotFoundException("Ocorreu um erro. A cozinha não foi encontrada.");
+        
+        Cozinha cozinha = cozinhaPorId.get();
+        
+        if (cozinhaRepository.existsByNomeIgnoreCaseAndIdNot(form.getNome(), Long.parseLong(id))) {
+        	model.addAttribute("erro", "Já existe uma cozinha com esse nome.");
+        	return editar(id, model);
+        }
+        
+    	cozinha.setNome(form.getNome());
+    	cozinhaRepository.save(cozinha);
+    	return "redirect:/cozinhas";
+    }
+
+    @PostMapping("/admin/cozinhas/excluir")
+    public String excluir(Long id) {
+        Optional<Cozinha> cozinha = this.cozinhaRepository.findById(id);
+        if (cozinha.isPresent()) this.cozinhaRepository.delete(cozinha.get());
```

Comment:
dá pra fazer

```suggestion
this.cozinhaRepository.findById(id).ifPresent(this.cozinhaRepository::delete);
```

Mas será que não deveria lançar o erro de não encontrado nesse caso também? :thinking: